doc/developer_guide/parsing.rst

```diff
@@ -137,6 +137,129 @@ create the appropriate `Arg` instance. Previously we relied on the
           extend if we were to support arithmetic operations in an
           invoke call.
 
+Mixed Precision
+===============
+
+Support for mixed precision kernels is being added to PSyclone but is
+still work-in-progress (see #1277). The approach being taken is for
+the user to provide kernels with a generic interface and
+precision-specific implementations. As the PSyclone kernel metadata
+does not specify precision, this does not need to change. The actual
+precision being used will be specified by the scientist in the
+algorithm layer (by declaring variables with appropriate precision).
+
+.. highlight:: fortran
+
+For example::
+
+    program alg
+      use kern_mod, only kern_type
+      real(32) :: a
+      real(64) :: b
+      call invoke(kern_type(a), kern_type(b))
+    end program alg
+
+    module kern_mod
+      type kern_type
+        ! metadata description
+	procedure :: kern_code
+      end type kern_type
+      interface kern_code
+	kern_code_32
```

Comment:
I don't think this is valid Fortran. I suggest `module procedure kern_code32, kern_code_64`

Reply:
You want valid Fortran! Shocking :-). OK, the example is now valid.